### PR TITLE
feat(rulesets): add new rule that requires sibling items field for type array

### LINKED ISSUE #2595: Wrongly declared array not caught as error
**Describe the bug**
The OAS file below has a wrongly declared array in the response. It states `type: array` and last line, and is missing an `items` key. Spectral does not catch this as an error. However, deploying such OAS in Apigee will not work as Apigeee uses Swagger parser which catches this error.

```yaml
openapi: 3.0.3
info:
  title: Wrong array
  description: Wrong array
  version: 1.0.0
  contact:
    name: IDA Support Team
    email: ida_team@maersk.com
    url: https://www.maersk.com/contactus
  license:
    url: https://terms.maersk.com/api-license-terms
    name: Maersk 1.0
servers:
  - url: https://api-stage.net
paths:
  /translatetoen:
    post:
      operationId: post-translation-request
      description: Sent a text for translation
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: object
              properties:
                inputText:
                  type: string
              required:
                - inputText
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  highestPrecendenceLanguage:
                    type: string
                  translatedText:
                    type: string
                  allIdentifiedLanguages:
                    type: array
                    items:
                      type: array
```

**To Reproduce**

1. Use above OAS, create file `oas-array-wrong.yaml`
2. Run CLI `spectral lint oas-array-wrong.yaml`
3. Output is `No results with a severity of 'error' found!`

**Expected behavior**
The wrongly declared array should be treated as an error as you cannot have an array declared without stating what type the items are.

**Screenshots**
NA

**Environment (remove any that are not applicable):**
 - Library version: 6.11.0
 - OS: MacOS
 - Browser: NA

### PR DESCRIPTION
**Pull request by rainum**
Fixes #2595.

This PR adds a new `array-items` rule for OAS 3.0.x which requires schemas with "type: array" to have a sibling "items" field.

There are some additional changes present that make tests more stable. 

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/rulesets/src/oas/index.ts [hunk 1]
```diff
@@ -361,6 +361,18 @@ const ruleset = {
         function: refSiblings,
       },
     },
+    'array-items': {
+      formats: [oas3_0],
```
Comment (daniel-white) [inline packages/rulesets/src/oas/index.ts:365]: should this be 3.1 as well?

File: packages/rulesets/src/oas/index.ts [hunk 2]
```diff
@@ -361,6 +361,18 @@ const ruleset = {
...
+      message: 'Schemas with "type: array", require a sibling "items" field',
+      severity: 0,
+      recommended: true,
+      resolved: false,
+      given: "$..[?(@.type === 'array')]",
+      then: {
+        function: truthy,
```
Comment (daniel-white) [inline packages/rulesets/src/oas/index.ts:372]: should this detect a child schema?

Comment (rainum) [reply packages/rulesets/src/oas/index.ts:372]: Good question! It should, I added a separate test for nested schemas: https://github.com/stoplightio/spectral/pull/2632/files#diff-7c7ca1323e671563e8696b6c8f4c6b7376a94a6cb93e08dfcec0774957b5a494R57-R63

File: packages/rulesets/src/oas/index.ts (hunk 1 above)
Comment (rainum) [reply packages/rulesets/src/oas/index.ts:365]: Yeah, we discussed this with @mnaumanali94. We agreed to fix one version at a time. We can address 3.1 if there will be any interest in the community.

**Review by daniel-white — DISMISSED** [review]

**Review by mnaumanali94 — DISMISSED** [review]
Looks good. @rainum Can we add a quick line here: https://github.com/stoplightio/spectral/blob/develop/docs/reference/openapi-rules.md

I can add it if you want.

**Comment by rainum** [comment]
> Looks good. @rainum Can we add a quick line here: https://github.com/stoplightio/spectral/blob/develop/docs/reference/openapi-rules.md

Oh, certainly! I completely forgot to do this, thanks for reminding me!

**Comment by rainum** [comment]
@mnaumanali94, I modified docs in 5bee34305381e9119da0a2d17ba7f69168a8ff0a 👌

**Comment by mnaumanali94** [comment]
LGTM!

**Review by mnaumanali94 — APPROVED** [review]

**Pull request merged**